lib/matplotlib/widgets.py

```diff
@@ -2588,14 +2571,12 @@ def __init__(self, ax, onselect, direction, *, minspan=0, useblit=False,
 
         self._active_handle = None
 
-    def new_axes(self, ax, *, _props=None):
+    def new_axes(self, ax, *, _props=None, _init=False):
         """Set SpanSelector to operate on a new Axes."""
         self.ax = ax
-        if self.canvas is not ax.figure.canvas:
+        if _init or self.canvas is not ax.figure.canvas:
```

Comment:
```suggestion
        if _init:
```

The second part will never be true now.

Reply:
It was pointed out I am wrong about being wrong.